tensorboard/plugins/text/text_plugin.py

```diff
@@ -164,29 +164,42 @@ def text_array_to_html(text_arr):
 
     Args:
       text_arr: A numpy.ndarray containing strings.
+      enable_markdown: boolean, whether to enable Markdown
 
     Returns:
       The array converted to html.
     """
     if not text_arr.shape:
-        # It is a scalar. No need to put it in a table, just apply markdown
-        return plugin_util.markdown_to_safe_html(text_arr.item())
+        # It is a scalar. No need to put it in a table.
+        if enable_markdown:
+            return plugin_util.markdown_to_safe_html(text_arr.item())
+        else:
+            return plugin_util.safe_html(text_arr.item())
     warning = ""
     if len(text_arr.shape) > 2:
         warning = plugin_util.markdown_to_safe_html(
             WARNING_TEMPLATE % len(text_arr.shape)
         )
         text_arr = reduce_to_2d(text_arr)
-    table = plugin_util.markdowns_to_safe_html(
-        text_arr.reshape(-1),
-        lambda xs: make_table(np.array(xs).reshape(text_arr.shape)),
-    )
+    if enable_markdown:
+        table = plugin_util.markdowns_to_safe_html(
+            text_arr.reshape(-1),
+            lambda xs: make_table(np.array(xs).reshape(text_arr.shape)),
+        )
+    else:
+        # Convert utf-8 bytes to str. The built-in np.char.decode doesn't work on
+        # object arrays, and converting to an numpy chararray is lossy.
+        decode = lambda bs: bs.decode("utf-8") if isinstance(bs, bytes) else bs
+        text_arr_str = np.array(
+            [decode(bs) for bs in text_arr.reshape(-1)]
+        ).reshape(text_arr.shape)
```

Comment:
Do note that the markdown version makes sure it accounts for null characters resulting from decoding mismatch. 

In tensorboard/plugin_util.py.
```py
            source_decoded = source.decode("utf-8")
            # Remove null bytes and warn if there were any, since it probably means
            # we were given a bad encoding.
            source = source_decoded.replace("\x00", "")
            total_null_bytes += len(source_decoded) - len(source)
...

    warning = ""
    if total_null_bytes:
        warning = (
            "<!-- WARNING: discarded %d null bytes in markdown string "
            "after UTF-8 decoding -->\n"
        ) % total_null_bytes
```

Reply:
Yeah, I considered adding that to `safe_html` but wasn't sure it was warranted since the sanitizer strips the null bytes anyway (confirmed with a [unit test](https://github.com/tensorflow/tensorboard/blob/ee9d15e15cf195eb4312dc4d3679c32cc1b9f934/tensorboard/plugin_util_test.py#L66-L74)). The reason given in the markdown version for doing the stripping is that otherwise, the markdown conversion gets confused by the null bytes: https://github.com/tensorflow/tensorboard/blob/ee9d15e15cf195eb4312dc4d3679c32cc1b9f934/tensorboard/plugin_util_test.py#L175-L181

So the only difference really is whether we insert the diagnostic HTML comment or not. That didn't quite seem worth any potential overhead to me, but I'm happy to add it back if you prefer?